### PR TITLE
Fixed Ueberauth request not respecting Script Name

### PR DESCRIPTION
**Pull request by evadne**
- When a router forwards traffic to another router, the conn will have it script name set.
- Read from it and override request / callback paths as well for good measure to accommodate umbrella structures.

This is a follow-up PR to address the scenario of per-app Ueberuath pipelines existing in an Umbrella app. Although configuration is loaded properly, in case the individual apps do not have individual Endpoints (instead, they just have Routers and a central Endpoint + Router forward traffic to them), then the incoming conn will have `script_name` set, which may or may not be reflected in configuration for each individual provider.

So for example, a Google Ueberauth provider will expect callbacks to hit `/auth/google/callback` by default although it may be hosted in an app whose Endpoint is getting forwarded traffic starting with `/myapp`. This set of changes in particular reads `script_name` dynamically and passes it on, so the provider gets paths that are correct and relative to root instead of relative to the root of the router.

### CODE REVIEW

File: lib/ueberauth.ex [hunk 1]
```diff
@@ -243,24 +243,33 @@ defmodule Ueberauth do
 
   @doc false
   def call(conn, routes) do
-    route_path = String.replace_trailing(conn.request_path, "/", "")
-    route_key = {route_path, conn.method}
-
+    route_prefix = Path.join(["/" | conn.script_name])
+    route_path = conn.request_path |> Path.relative_to(route_prefix)
```
Comment (doomspork) [inline lib/ueberauth.ex]: Could we remove the pipeline here since there's only a single function call?

**Comment by yordis** [comment]
@evadne this definitely will improve over what we have but what we should fix the underline implementation to be more Plug-less based.

The issue today is that Ueberauth does not allow us to pass configurations at runtime (I mean `not` reading from `config/*.exs` files) because of this issue, umbrella apps or use cases where I want to swap the configuration based on the connection is not possible.

There is an effort to fix this in one of the branches, basically, it allows you to initialize the request phase passing the configuration.

This will allow us to refactor the plugs and the other layer where we can make assumptions of like this one where you could use the application name to read the config, or the default one 🤷‍♂️ (whatever is the use case)

But before we introduce more complexity I believe we should fix the issue with configurations in general and then come back to this.

**Comment by doomspork** [comment]
@evadne could you rebase on the latest master? There should be some travis changes.

**Comment by evadne** [comment]
@doomspork rebased now

I will be happy to fix this further as required

**Comment by doomspork** [comment]
Sorry for the delay @evadne, I've been taking care of personal things and it's left little time for open source things.

**Comment by yordis** [comment]
Nope

**Comment by evadne** [comment]
Could we cut another release for this one? Thanks

**Comment by yordis** [comment]
@evadne hopefully I don't forget once I get home today!

**Pull request merged**